File: 2502-sort-the-people/sort-the-people.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        vector<int> indices(names.size());
        iota(indices.begin(), indices.end(), 0);

        sort(indices.begin(), indices.end(), [&](int lhs, int rhs) {
            return heights[lhs] > heights[rhs];
        });

        vector<string> sortedNames(names.size());
        for (int i = 0; i < names.size(); ++i) {
            sortedNames[i] = names[indices[i]];
        }

        return sortedNames;
    }
};

File: 2502-sort-the-people/sort-the-people.swift
class Solution {
    func sortPeople(_ names: [String], _ heights: [Int]) -> [String] {
        let n = names.count
        var indices: [Int] = Array(0..<n)

        indices = indices.sorted(by: {lhs, rhs in
            heights[lhs] > heights[rhs]
        })

        var sortedNames: [String] = []
        for idx in 0..<n {
            sortedNames.append(names[indices[idx]])
        }

        return sortedNames
    }
}